### PR TITLE
Remove HTTP from supported protocols in Downloader

### PR DESCRIPTION
**Pull request by OleksandrChaika**

### CODE REVIEW

File: utest/TestSelfUpdateScenarios.cpp [hunk 1]
```diff
@@ -409,4 +411,21 @@ namespace {
         EXPECT_EQ(sentMessages, expectedMessages);
     }
 
+    TEST_F(TestSelfUpdateScenarios, downloadFromHTTPFails_endsInFailedState)
```
Comment (OlavHerbst) [inline utest/TestSelfUpdateScenarios.cpp]: rename test like "downloadViaHttpConnectionFails_endsInFailedState"
(in CamelCase names contained acronyms should be written in lowercase to be able to better separate the words while reading)

Comment (OlavHerbst) [inline utest/TestSelfUpdateScenarios.cpp]: Missing TCs for other protocols to be also failing, at least ftp, sftp, smb, smbs, ldap, ldaps
Ok, the task says only "Configure libcurl to not use HTTP protocol", but curl config rejects all protocols now except https. So more protocols should also be tested? (Can't be complete.)

**Review by OlavHerbst — APPROVED** [review]
ok, wrote just a suggestion

Req. says "download is rejected by SUA and feedback is written to log "
Missing log output.

File: utest/TestSelfUpdateScenarios.cpp (hunk 1 above)
Comment (xvoidee) [reply utest/TestSelfUpdateScenarios.cpp:414]: Fixed

Comment (OleksandrChaika) [reply utest/TestSelfUpdateScenarios.cpp]: Fixed

Comment (OleksandrChaika) [reply utest/TestSelfUpdateScenarios.cpp]: Added parameterized test case

File: src/Download/Downloader.cpp [hunk 2]
```diff
@@ -103,50 +136,49 @@ namespace {
...
         fclose(fp);
         progressNotificationLimiter = 0;
 
         sua::Logger::debug("CURLINFO_RESPONSE_CODE = {}", http_code);
         if(http_code != 200) {
-            sua::Logger::error(curl_easy_strerror(res));
-            return sua::TechCode::DownloadFailed;
+            auto e = curl_easy_strerror(res);
+            sua::Logger::error(e);
+            return std::make_tuple(sua::TechCode::DownloadFailed, "");
```
Comment (OlavHerbst) [inline src/Download/Downloader.cpp]: would add "Errorcode = ...", to get later not an error message with only "Download failed:"

File: utest/TestDownloader.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        auto res = d.start(url);
+
+        EXPECT_EQ(std::get<0>(res), sua::TechCode::DownloadFailed);
+    }
+
+    INSTANTIATE_TEST_CASE_P(
+        TestDownloaderViaUnsupportedProtocols,
+        TestDownloaderP,
+        ::testing::Values(
+            "ftp", "sftp", "smb", "smbs", "ldap", "ldaps"
```
Comment (OlavHerbst) [inline utest/TestDownloader.cpp]: missing "http"

**Review by OlavHerbst — CHANGES_REQUESTED** [review]
missing test case for "http"

**Review by OlavHerbst — CHANGES_REQUESTED** [review]
2 findings

File: utest/TestDownloader.cpp (hunk 3 above)
Comment (OleksandrChaika) [reply utest/TestDownloader.cpp]: Fixed

File: src/Download/Downloader.cpp (hunk 2 above)
Comment (OleksandrChaika) [reply src/Download/Downloader.cpp]: Added error to 2nd value in tuple

**Review by OlavHerbst — APPROVED** [review]
ok

**Review by OlavHerbst — APPROVED** [review]
ok

**Pull request merged**